src/dscanner/main.d

```diff
@@ -423,8 +422,14 @@ string getDefaultConfigurationLocation()
 			configDir = buildPath(configDir, "dscanner", CONFIG_FILE_NAME);
 		return configDir;
 	}
-	else version (Windows)
-		return CONFIG_FILE_NAME;
+    else version(Windows)
+    {
+        string configDir = environment.get("USERPROFILE", null);
+        if (configDir is null)
+            throw new Exception("$USERPROFILE is unset");
```

Comment:
FWIW we could simply do:

```
environment.get("USERPROFILE");
```

if all we want is an exception if the variable doesn't exist.

Reply:
That the same thing, the default value passed i just a null, it only throws if there is an UTF error. According to the Doc at least.

https://github.com/dlang/phobos/blob/v2.079.0/std/process.d#L3375

Just a small nitpick, `$USERPROFILE` would be the bash way of declaring a variable. The windows way is either `$env:USERPROFILE` for power shell or `%USERPROFILE%` for cmd. I imagine most people will know what it means just by virtue of the name and caps.

Reply:
there's opIn now so finally you can do

`if (const string* = "USERPROFILE" in environment) {...}`

Reply:
```diff
-        string configDir = environment.get("USERPROFILE", null);
+       string configDir = environment.get("APPDATA", null);
        if (configDir is null)
            throw new Exception("%APPDATA% is unset");
-        configDir = buildPath(configDir, ".config", "dscanner", CONFIG_FILE_NAME);
+        configDir = buildPath(configDir, "dscanner", CONFIG_FILE_NAME);
        return configDir;
```